server/service/grpc/service.go

```diff
@@ -170,17 +170,33 @@ func (s *Service) CreateTable(ctx context.Context, req *metaservicepb.CreateTabl
 		return nil
 	}
 
-	procedure, err := factory.CreateCreateTableProcedure(ctx, procedure.CreateTableRequest{
-		Cluster:     c,
-		SourceReq:   req,
-		OnSucceeded: onSucceeded,
-		OnFailed:    onFailed,
-	})
-	if err != nil {
-		log.Error("fail to create table", zap.Error(err))
-		return &metaservicepb.CreateTableResponse{Header: responseHeader(err, "create table")}, nil
+	var p procedure.Procedure
+	if req.PartitionInfo != nil && len(req.PartitionInfo.GetNames()) != 0 {
```

Comment:
`if req.PartitionInfo != nil && len(req.PartitionInfo.GetNames()) == 0`, shall we return error?
Shall we paas `CreateTableRequest` to `factory`, `factory` decides  to create `CreatePartitionTableProcedure` or `CreateTableProcedure`.

Reply:
I guess it is not good to do validation for `CreateTableRequest` in the service layer because the service layer have no idea about the specific logic about what is necessary to create table and another validation is still required in the procedure layer even if it is done by the service layer.

However, it is not good to let the `factory` to take over such validation because of the same reason: the `factory` should not have any knowledge about that.

So I guess the best way is to let the specific procedure to do such validation, e.g. `CreatePartitionTableProcedure`.

Reply:
I guess it is not a good idea to decide which procedure to create in the service layer because service layer has no knowledge about how to create table.

I guess we can make this decision in `CreateTableProcedure`, that is to say, the procedure factory has no idea about the `CreatePartitionTableProcedure`, which is just a creating branch in the `CreateTableProcedure`. And there are still two different specific procedures according to the `CreateTableRequest`.

Reply:
Solve this problem by adding a proxy procedure.